app/views/activities/order_screen.html.erb

```diff
@@ -176,29 +191,32 @@
               Sumup betaling
             </span>
           </a>
-          <button type="button" class="btn btn-dark" v-bind:disabled="orderConfirmButtonDisabled()"
+          <button type="button" class="btn btn-dark" :disabled="orderConfirmButtonDisabled()"
                   @click="confirmOrder">
             <%= fa_icon 'money' %>
           </button>
         </div>
 
-        <button @click="confirmOrder" class="btn btn-block btn-lg btn-success"
-                :class="{ 'btn-warning' : (orderRequiresAge() && selectedUser && selectedUser.minor) }"
-                :disabled="orderConfirmButtonDisabled()"
-                v-if="!payWithPin || !<%= @sumup_enabled.to_s %>">
-          <span class="text-uppercase">
-            Bestelling plaatsen
+        <button @click="maybeConfirmOrder" class="btn btn-block btn-lg btn-success"
+              :class="{ 'btn-warning' : selectedUser && ((orderRequiresAge() && selectedUser.minor) || selectedUser.insufficient_credit) }"
+              :disabled="orderConfirmButtonDisabled()"
+              v-if="!payWithPin || !<%= @sumup_enabled.to_s %>">
+        <span class="text-uppercase">
+          Bestelling plaatsen
+        </span>
+        <br/>
+        <small v-if="selectedUser && (selectedUser.insufficient_credit || (orderRequiresAge() && selectedUser.minor))">
```

Comment:
Somewhat complex condition, and parts of this are reused at certain places. Consider extracting checks like these to computed props in the future, as it simplifies templates and clarifies conditions using identifiers

Reply:
I have extracted them to computed properties